### PR TITLE
[scripts] Create internal-dependency-trace script package

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
Adds a new package under `scripts/internal-dependency-trace` that can be used to generate a rendering of any internal file dependency graph.

Default configured to generate the directed dependency graph of `packages/cli/src/index.ts`

### CODE REVIEW

File: scripts/internal-dependency-trace/readme.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Internal Dependency Trace
+
```
Comment (EndangeredMassa) [inline scripts/internal-dependency-trace/readme.md:2]:
```suggestion

This script will generate a [Mermaid](https://mermaid.js.org/intro/) directed graph as an SVG image showing the dependency connections between files starting with the input file as an entrypoint.

```

**Review by EndangeredMassa — DISMISSED** [review]

**Review by cb1kenobi — DISMISSED** [review]

**Review by EndangeredMassa — APPROVED** [review]

**Review by styfle — COMMENTED** [review]
@Ethan-Arrowood How is this different than turborepo?

You can generate mermaid with

```sh
turbo run build --graph=mygraph.mermaid
```

Or render the mermaid to graph to html with

```sh
turbo run build --graph=mygraph.html
```

See https://turbo.build/repo/docs/reference/command-line-reference/run#--graph

**Comment by Ethan-Arrowood** [comment]
Not sure - I think when I added this maybe we didn't realize Turbo could do it.

**Comment by Ethan-Arrowood** [comment]
And also, this like traces the direct files so I could better understand the web of the repo. Turbo might stop at literal package dependencies

**Pull request merged**